spec/README.md

```diff
@@ -0,0 +1,172 @@
+
+```go
+type RawLeaf struct {
+	DepositAddress     string
+	AccumulatedBalance *big.Int
+}
+```
+
+TODO:
+https://excalidraw.com/#json=B9-GYxQulDA5PA-xanxwP,-D8sGwTKVlykXXLiLZPpeA
+
+* The merkle tree leafs are ordered by its deposit address in ascending order.
+* The merkle tree is prepended (first element) with a leaf containing `POOL_FEES_ADDRESS` and the accumulated balance. See rewards calculation section.
+* The merkle leafs are hashed with solidity `sha3` hashing algorithm.
+* Deposit addresses in the merkle tree shall be unique.
+* The hashing algorithm for the merkle tree is `keccak256`.
+* The hashing algotithm for the merkle tree shall sort sibling pairs.
+* The deposit addresses in the merkle tree shall be in lower case.
+
+See test vectors for more information. TODO:
+
+Every `CHECKPOINT_SIZE_SLOTS` the oracle updates in the smoothing pool smart contract stored in the Ethereum blockchain a new merkle root, that summarizes the rewards that each address can claim. Anyone that controls said address, can claim their rewards by providing a valid merkle proof, prooving that a given leaf is contained within the merkle tree represented by that merkle root.
+
+Since all this data is not available in Ethereum, the oracle shall provide this proofs so that they can be used off-chain. Note that these proofs can be generated by anyone compliying with this specs and with the existing available data on-chain. See [merkle proofs](https://ethereum.org/es/developers/tutorials/merkle-proofs-for-offline-data-integrity/)
+
+
+## Smart contract
+
+TODO: Write some notes on the smart contract.
+https://github.com/dappnode/mev-sp-contracts
+
+### Test vectors
+
+* TODO: test vector for merkle trees, proofs, hashes, roots.
+* TODO: test vector for rewards calculation + collaterals + fees
+
+## Oracle api
+
+Optionally, the oracle can have an api to get the state of the validators such as balances, merkle proofs, merkle root, etc. Some of the recommended endpoints are:
+
+TODO: Add doc on this
+
+curl 157.90.93.245:7300/status (returns status)
+curl 157.90.93.245:7300/depositadddress/459438 (returns deposit address of a given validator index)
+curl 157.90.93.245:7300/proof/0x005CD1608e40d1e775a97d12e4f594029567C071 (returns the merkle proof of said deposit address for latest onchain state)
+curl 157.90.93.245:7300/validatoronchainstate/179434 (return the state of the validator index)
+
+TODO: Get latest onchain merkle root
+TODO:
+
+## FAQ
+TODO:
```

Comment:
Q: cant see my validator?
A: Validators are updated every x hours and must be active (not pending to be activaed). Slashed and exited validators are not displayed.